docs/how-delta-lake-works/delta-lake-acid-transactions.md

```diff
@@ -0,0 +1,269 @@
+
+These guarantees are provided at the protocol level through the "transaction" abstraction.  We’ve already discussed what constitutes a transaction for Delta Lake.
+
+If there is an error with the transaction and some files don’t get written, then no metadata entry is made and the partial data write is ignored.  The zombie Parquet files can be easily cleaned up via subsequent vacuum operations.
+
+Now let’s look at how Delta Lake also provides consistent transactions.
+
+**Consistent transactions**
+
+Consistency means that transactions won’t violate integrity constraints on the Delta table.
+
+Delta Lake has two types of consistency checks:
+
+* Schema enforcement checks
+* Column constraints
+
+Schema enforcement checks verify that new data appended to a Delta table matches the schema of the existing table.  You cannot append data with a different schema, unless you enable schema evolution.
+
+Delta Lake column constraints allow users to specify the requirements of data that’s added to a Delta table.  For example, if you have an age column with a constraint that requires the value to be positive, then Delta Lake will reject appends of any data that doesn’t meet the constraint.
+
+Data lakes don’t support schema enforcement or column constraints.  That’s another reason why data lakes are not ACID-compliant.
+
+**Isolated transactions**
+
+Isolation means that transactions are applied to a Delta table sequentially.
+
+Delta Lake transactions are persisted in monotonically increasing transaction files, as we saw in the previous example.  First `00000000000000000000.json`, then `00000000000000000001.json`, then `00000000000000000002.json`, and so on.
+
+Delta Lake uses concurrency control to ensure that transactions are executed sequentially, even when user operations are performed concurrently.  The next page of this guide explains concurrency in Delta Lake in detail.
+
+**Durable transactions**
+
+Delta tables are generally persisted in cloud object stores which provide durability guarantees.
+
+Durability means that all transactions that are successfully completed will always remain persisted, even if there are service outages or program crashes.
+
+Suppose you have a Delta table that’s persisted in Azure blog storage.  The Delta table transactions that are committed will always remain available, even in these circumstances:
+
+* When there are Azure service outages
+* If a computation cluster that’s writing the Delta table crashes for some reason
+* Two operations are running concurrently and one of them fails
+
+Successful transactions are always registered in the Delta table and persisted no matter what.
+
+## Conclusion
+
+Delta Lake supports transactions which provide necessary reliability guarantees for production data systems.
+
+Data lakes don’t provide transactions and this can cause nasty bugs and a bad user experience.  Let’s look at a couple of scenarios when the lack of transactions cause a poor user experience:
```

Comment:
Maybe call it vanilla data lakes? Delta lake in my opinion is just a data lake with a  metadata layer

Reply:
Updated this.  Delta Lakes def aren't data lakes 😱 

See [the Lakehouse paper](https://www.cidrdb.org/cidr2021/papers/cidr2021_paper17.pdf).  We usually call these "Lakehouse storage systems" or "open table formats".